tool.py

```diff
@@ -0,0 +1,154 @@
+# specific language governing permissions and limitations
+# under the License.
+
+# -*- coding: utf-8 -*-
+"""Tool to ease working with the build system and reproducing test results"""
+
+import os
+import sys
+from subprocess import check_call
+import shlex
+from ci.util import retry, remember_cwd
+from typing import List
+from collections import OrderedDict
+import logging
+import yaml
+
+class Confirm(object):
+    def __init__(self, cmds):
+        self.cmds = cmds
+
+    def __call__(self):
+        resp = input("This will run the following command(s) '{}' are you sure? yes / no: ".format(self.cmds))
+        while True:
+            if resp.lower() == 'yes':
+                handle_commands(self.cmds)
+                return
+            elif resp.lower() == 'no':
+                return
+            else:
+                resp = input("Please answer yes or no: ")
+
+class CMake(object):
+    def __init__(self, cmake_options_yaml='cmake/cmake_options.yaml'):
+        self.cmake_options_yaml = cmake_options_yaml
+        self.cmake_options = None
+        self.read_config()
+
+    def read_config(self):
+        assert os.path.isfile(self.cmake_options_yaml)
+        with open(self.cmake_options_yaml, 'r') as f:
+            self.cmake_options = yaml.load(f)
+
+    def _cmdlineflags(self):
+        res = []
+        def _bool_ON_OFF(x):
+            if x:
+                return 'ON'
+            else:
+                return 'OFF'
+        for opt,v in self.cmake_options.items():
+            res.append('-D{}={}'.format(opt,_bool_ON_OFF(v)))
+        return res
+
+    def cmake_command(self) -> str:
+        """
+        :return: Cmake command to run given the options
+        """
+        cmd_lst = ['cmake']
+        cmd_lst.extend(self._cmdlineflags())
+        return cmd_lst
+
+    def __call__(self, build_dir='build', generator='Ninja', build_cmd='ninja'):
+        logging.info("CMake / {} build in directory {}".format(
+            generator, os.path.abspath(build_dir)))
+        cmd_lst = self.cmake_command()
+        os.makedirs(build_dir, exist_ok=True)
+        with remember_cwd():
+            os.chdir(build_dir)
+            cmd_lst.extend(['-G{}'.format(generator), '..'])
+            logging.info('Executing: {}'.format('\t\n'.join(cmd_lst)))
+            check_call(cmd_lst)
+            logging.info('Now building')
+            check_call(shlex.split(build_cmd))
+
+
+
+COMMANDS = OrderedDict([
+    ('sanity_check',
+        "ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh sanity_check"),
+    ('Python3 CPU unittests',
+    [
+        "ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh build_ubuntu_cpu_openblas",
+        "ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh unittest_ubuntu_python3_cpu",
+    ]),
+    ('Python3 GPU unittests',
+    [
+        "ci/build.py --platform ubuntu_gpu /work/runtime_functions.sh build_ubuntu_gpu_cuda91_cudnn7",
```

Comment:
We might want to make this one dynamic because this would break if we rename the function due to cuda upgrade?